src/main/java/org/opensearch/agent/tools/PPLTool.java

```diff
@@ -83,12 +87,46 @@ public class PPLTool implements Tool {
 
     private String contextPrompt;
 
+    private PPLModelType pplModelType;
+
     private static Gson gson = new Gson();
 
-    public PPLTool(Client client, String modelId, String contextPrompt) {
+    private static Map<String, String> defaultPromptDict;
+
+    static {
+        try {
+            defaultPromptDict = loadDefaultPromptDict();
+        } catch (IOException e) {
+            throw new RuntimeException("fail to load default prompt dict" + e.getMessage());
```

Comment:
Is it possible that this would break the plugin initialization process with this error thrown?

Reply:
You're right. Already change the logic to init the dict to an empty dict and log an error here.